protocol/transport/transport.go

```diff
@@ -736,3 +736,35 @@ func (t *Transport) ConfirmMessageDelivered(messageID string) {
 func (t *Transport) SetStorePeerID(peerID peer.ID) {
 	t.waku.SetStorePeerID(peerID)
 }
+
+func (t *Transport) CheckForMissingMessages(peerID peer.ID, filters []*Filter) {
+	if t.waku.Version() != 2 {
+		return
+	}
+
+	topicMap := make(map[string]map[string]struct{})
+	for _, f := range filters {
```

Comment:
Would be better to move this logic further down in the stack as it is waku specific. Maybe part of missing_messages.go under wakuv2.